VeterinaryVladimir/Program.cs

```diff
@@ -8,8 +8,231 @@ namespace VeterinaryVladimir
+                if (cki.Key.ToString() == "H" || cki.Key.ToString() == "h")
+                {
+                    Console.Clear();
+                    Console.SetCursorPosition(0, 0);
+                    Console.WriteLine("Welcome!");
+                    Console.ReadKey();
+                }
+                else if (cki.Key.ToString() == "R" || cki.Key.ToString() == "r")
+                {
+                    // ---MenuR();  // Добавить элемент
+                    ConsoleKeyInfo ckiR;
+                    do
+                    {
+                        int ageAnimal = 0;
+                        Console.Clear();
+                        Console.SetCursorPosition(0, 0);
+
+                        Console.WriteLine("Enter for add: [C] - Cat, [D] Dog, [H] Hamster, [F] Fish, [Q] Quit");
+                        Console.WriteLine("[C] Cat");
+                        Console.WriteLine("[D] Dog");
+                        Console.WriteLine("[H] Hamster");
+                        Console.WriteLine("[F] Fish");
+                        Console.WriteLine("[Q] Quit");
+
+                        ckiR = Console.ReadKey();
+                        if (ckiR.Key.ToString() == "C" || ckiR.Key.ToString() == "c")
+                        {
+                            Console.Clear();
+                            Console.SetCursorPosition(0, 0);
+                            Console.WriteLine("Name of Cat?");
+                            string nameAnimal = "0";
+                            nameAnimal = Console.ReadLine();
+                            do
+                            {
+                                Console.WriteLine("Age of Cat? Only int!");
+                            }
+                            while (!int.TryParse(Console.ReadLine(), out ageAnimal) || ageAnimal < 0);
+                            HomeCat enimal = new HomeCat(nameAnimal, ageAnimal);
+                            clinica.addAnimal(enimal);
+                        }
+                        else if (ckiR.Key.ToString() == "D" || ckiR.Key.ToString() == "d")
+                        {
+                            Console.Clear();
+                            Console.SetCursorPosition(0, 0);
+                            Console.WriteLine("Name of Dog?");
+                            string nameAnimal = "0";
+                            nameAnimal = Console.ReadLine();
+                            do
+                            {
+                                Console.WriteLine("Age of Dog? Only int!");
+                            }
+                            while (!int.TryParse(Console.ReadLine(), out ageAnimal) || ageAnimal < 0);
+                            HomeDog enimal = new HomeDog(nameAnimal, ageAnimal);
+                            clinica.addAnimal(enimal);
+                        }
+                        else if (ckiR.Key.ToString() == "H" || ckiR.Key.ToString() == "h")
+                        {
+                            Console.Clear();
```

Comment:
I don't like all these repeated blocks with single different "Name of..."  and "Age of...".
Can you propose a solution without duplication of similar pieces of code?